### DISCUSSION TITLE
Error when installing with NFS mount as Datadir

### DISCUSSION THREAD

**Discussion by LeoSum8** [nextcloud/all-in-one#3529] [Questions] [answered]
I am trying to install Nextcloud AIO on my homeserver. It is a TrueNAS system, running a bhyve VM as docker-host, which is running Fedora CoreOS (FCOS). All relevant data is stored in the ZFS-Filesystem of the TrueNAS host, mounted into the FCOS VM via NFS-shares and then bind-mounted in to the docker containers. This procedure is working well for a bunch of containers that I am running on the system.

A first test of Nextcloud-AIO, where I just used an empty local folder on the VM's filesystem to act as nextcloud-data dir (via the NEXTCLOUD_DATADIR env variable) worked fine.

However now that I am trying to set up my "production" setup involving the NFS-share as nextcloud-data dir, I get the following error when downloading and starting the containers:

```
Trying to fix docker.sock permissions internally...
Creating docker group internally with id 982
.+...+.+++++++
-----
[0;92mInitial startup of Nextcloud All-in-One complete!
You should be able to open the Nextcloud AIO Interface now on port 8080 of this server!
E.g. https://internal.ip.of.this.server:8080
If your server has port 80 and 8443 open and you point a domain to your server, you can get a valid certificate automatically by opening the Nextcloud AIO Interface via:
https://your-domain-that-points-to-this-server.tld:8443[0m
{"level":"info","ts":1696621970.1793003,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
[06-Oct-2023 19:52:50] NOTICE: fpm is running, pid 133
[06-Oct-2023 19:52:50] NOTICE: ready to handle connections
[Fri Oct 06 19:52:50.182408 2023] [mpm_event:notice] [pid 124:tid 140475841915720] AH00489: Apache/2.4.57 (Unix) OpenSSL/3.1.3 configured -- resuming normal operations
[Fri Oct 06 19:52:50.182554 2023] [core:notice] [pid 124:tid 140475841915720] AH00094: Command line: 'httpd -D FOREGROUND'
NOTICE: PHP message: It seems like the ip-address of nextcloud.example.com is set to an internal or reserved ip-address. (It was found to be set to '192.168.178.137')
NOTICE: PHP message: Slim Application Error
Type: GuzzleHttp\Exception\ServerException

0
Message: Server error: `POST http://localhost/v1.41/containers/nextcloud-aio-nextcloud/start` resulted in a `500 Internal Server Error` response:
{&quot;message&quot;:&quot;failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process (truncated...)
File: /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php

3
Trace: #0 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Middleware.php(72): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL)
#1 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(209): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response))
#2 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(158): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL)
#3 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/TaskQueue.php(52): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
#4 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(251): GuzzleHttp\Promise\TaskQueue->run(true)
#5 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(227): GuzzleHttp\Promise\Promise->invokeWaitFn()
#6 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(272): GuzzleHttp\Promise\Promise->waitIfPending()
#7 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(229): GuzzleHttp\Promise\Promise->invokeWaitList()
#8 /var/www/docker-aio/php/vendor/guzzlehttp/promises/src/Promise.php(69): GuzzleHttp\Promise\Promise->waitIfPending()
#9 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Client.php(189): GuzzleHttp\Promise\Promise->wait()
#10 /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/ClientTrait.php(95): GuzzleHttp\Client->request('POST', 'http://localhos...', Array)
#11 /var/www/docker-aio/php/src/Docker/DockerActionManager.php(189): GuzzleHttp\Client->post('http://localhos...')
#12 /var/www/docker-aio/php/src/Controller/DockerController.php(48): AIO\Docker\DockerActionManager->StartContainer(Object(AIO\Container\Container))
#13 /var/www/docker-aio/php/src/Controller/DockerController.php(33): AIO\Controller\DockerController->PerformRecursiveContainerStart('nextcloud-aio-n...', true)
#14 /var/www/docker-aio/php/src/Controller/DockerController.php(193): AIO\Controller\DockerController->PerformRecursiveContainerStart('nextcloud-aio-a...', true)
#15 /var/www/docker-aio/php/src/Controller/DockerController.php(173): AIO\Controller\DockerController->startTopContainer(true)
#16 /var/www/docker-aio/php/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(38): AIO\Controller\DockerController->StartContainer(Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
#17 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(358): Slim\Handlers\Strategies\RequestResponse->__invoke(Array, Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
#18 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Slim\Routing\Route->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#19 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#20 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(315): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#21 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/RouteRunner.php(68): Slim\Routing\Route->run(Object(GuzzleHttp\Psr7\ServerRequest))
#22 /var/www/docker-aio/php/vendor/slim/csrf/src/Guard.php(476): Slim\Routing\RouteRunner->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#23 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(168): Slim\Csrf\Guard->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Slim\Routing\RouteRunner))
#24 /var/www/docker-aio/php/vendor/slim/twig-view/src/TwigMiddleware.php(115): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#25 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Views\TwigMiddleware->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#26 /var/www/docker-aio/php/src/Middleware/AuthMiddleware.php(38): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#27 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(269): AIO\Middleware\AuthMiddleware->__invoke(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#28 /var/www/docker-aio/php/vendor/slim/slim/Slim/Middleware/ErrorMiddleware.php(76): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#29 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Middleware\ErrorMiddleware->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#30 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#31 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(199): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#32 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(183): Slim\App->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#33 /var/www/docker-aio/php/public/index.php(182): Slim\App->run()
#34 {main}
Tips: To display error details in HTTP response set "displayErrorDetails" to true in the ErrorHandler constructor.
```

This happens with both the latest and the beta tag.

I am starting via:

```
docker run \
--init \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 8999:8080 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
--env APACHE_IP_BINDING=127.0.0.1 \
--env APACHE_PORT=11000 \
--env NEXTCLOUD_DATADIR="/mnt/nextcloud_files_nfs/" \
--env NEXTCLOUD_MOUNT="/mnt/" \
nextcloud/all-in-one:beta
```

Any ideas on why that is happening or how to narrow down the issue?
I tried different combinations and this only happens when NEXTCLOUD_DATADIR is in a folder of an NFS mount. However the `--env NEXTCLOUD_MOUNT="/mnt/"` works and I can use nfs-mounts as external storage that way.

The issue seems kinda similar to [this one](https://github.com/nextcloud/all-in-one/discussions/1747), but I'd rule out a PID limit in my case.

**Comment by szaimen** [comment]
> failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process

I've found this: https://www.google.com/search?q=%22failed%20to%20create%20shim%20task%3A%20OCI%20runtime%20create%20failed%3A%20runc%20create%20failed%3A%20unable%20to%20start%20container%20process%22

**Comment by LeoSum8** [comment]
Thank you for looking into my issue and thank you for not linking to https://letmegooglethat.com/ ;)

I had already done some searching, but the google results don't seem to match my case. But one could lead the right way:

1. https://ao.bloatcat.tk/questions/72695311/failure-starting-docker-container-failed-to-create-shim-task-oci-runtime-crea
    - Root Cause: compose timeout
    - Suggested Solution: increase the compose timeout (COMPOSE_HTTP_TIMEOUT=480)
    -> Maybe in my case the NFS is too slow? Do you use compose internally in the master container? How would I set this increased timeout?

2. https://code.whatever.social/questions/75078744/docker-error-response-from-daemon-failed-to-create-shim-task-oci-runtime-crea
    - Root Cause: Execution Permission missing for a file
    - Suggested Solution: chmod +x
    -> Not related

3. https://code.whatever.social/questions/75078744/docker-error-response-from-daemon-failed-to-create-shim-task-oci-runtime-crea
    - Root Cause: wrong command formatting in docker compose file 
    - Suggested Solution: adjust compose file
    -> Not related
...

Another thing I tried is this:
`sudo docker run -it -v /mnt/nextcloud_files_nfs:/mnt/ncdata --entrypoint /bin/sh nextcloud/aio-nextcloud:beta`
From inside the container I can then read and write inside /mnt/ncdata without any problems.

Also I have disabled SELinux already.

Do you understand at what step things are going wrong here?
It looks like the startup of the nextcloud-aio-nextcloud container is triggered by a POST to some API and that returns said error.
Do I have an option to get some more verbose output or retry the steps manually to see what is happening then?

the log output of the nextcloud-aio-nextcloud container is empty btw since it is never started.

**Reply by bradley-rose** [reply]
Did you ever sort this out? I'm running into the exact same issue.

**Reply by LeoSum8** [reply]
I think I had wrongfully added trailing slashes to the path for the ncdata volume in the docker run statement. And I think it went away after starting Form scratch without that slash. But its a long time ago. Sorry

**Comment by LeoSum8** [comment]
Would it make sense to try the manual install to see if it brings up more info on the error?
https://github.com/nextcloud/all-in-one/tree/main/manual-install

**Comment by szaimen** [comment, marked as answer]
> A first test of Nextcloud-AIO, where I just used an empty local folder on the VM's filesystem to act as nextcloud-data dir (via the NEXTCLOUD_DATADIR env variable) worked fine.
> 
> However now that I am trying to set up my "production" setup involving the NFS-share as nextcloud-data dir, I get the following error when downloading and starting the containers:

Based on that it points towards an issue by using NFS as datadir so I am pretty sure that it is not caused by AIO as it works with normal block storage. Also I cannot reproduce your issue on my test instance (I am also not using NFS as datadir and cannot do so).

I would recommend to get some help on this in a Docker/NFS forum which seems to cause the issue
